akka-actor/src/main/scala/akka/pattern/RetrySupport.scala

```diff
@@ -30,17 +30,73 @@ trait RetrySupport {
    * private val sendReceiveRetry: HttpRequest => Future[HttpResponse] = (req: HttpRequest) => retry[HttpResponse](
    *   attempt = () => sendAndReceive(req),
    *   attempts = 10,
+   *   delay = 2 seconds)(
+   *    ec = context.system
+   *    scheduler = context.system.scheduler
+   * )
+   * }}}
+   */
+  def retry[T](attempt: () ⇒ Future[T], attempts: Int, delay: FiniteDuration)(implicit ec: ExecutionContext, scheduler: Scheduler): Future[T] =
+    retryF(attempt, attempts, delay, 1)((_, _) ⇒ ())
+
+  /**
+   * Given a function from Unit to Future, returns an internally retrying Future
+   * The first attempt will be made immediately, each subsequent attempt will be made after 'delay' * 'backoff'
+   * A scheduler (eg context.system.scheduler) must be provided to delay each retry
+   * If attempts are exhausted the returned future is simply the result of invoking attempt.
+   * Note that the attempt function will be invoked on the given execution context for subsequent
+   * tries and therefore must be thread safe (not touch unsafe mutable state).
+   *
+   * <b>Example usage:</b>
+   *
+   * {{{
+   * protected val sendAndReceive: HttpRequest => Future[HttpResponse]
+   * private val sendReceiveRetry: HttpRequest => Future[HttpResponse] = (req: HttpRequest) => retry[HttpResponse](
+   *   attempt = () => sendAndReceive(req),
+   *   attempts = 10,
+   *   delay = 2 seconds,
+   *   backoff = 2)(
+   *    ec = context.system
+   *    scheduler = context.system.scheduler
+   * )
+   * }}}
+   */
+  def retry[T](attempt: () ⇒ Future[T], attempts: Int, delay: FiniteDuration, backoff: Int)(implicit ec: ExecutionContext, scheduler: Scheduler): Future[T] =
```

Comment:
should the backoff be a `Double`?

Reply:
It could be, but multiplying a `FiniteDuration` with a `Double` will return a `Duration`. The current `FutureTimeoutSupport.after` method that we're using does expect a `FiniteDuration` so passing in a `Double` will lead to some potentially unsafe conversions.

And besides that, i think a `Double` backoff will make it harder to reason about the code. It's one thing to backoff by a factor of 2 and a totally different thing to backoff by 3.7... 😄

Reply:
I think we should use the same parameters as in other places where exponential backoff is used, e.g.
https://github.com/akka/akka/blob/master/akka-actor-typed/src/main/scala/akka/actor/typed/SupervisorStrategy.scala#L124

The factor is often not interesting to adjust. Using 2 is fine.

Reply:
@patriknw I see what you mean with the parameters...
I tried as much as possible to keep this function and its parameters similar to the other `retry` functions in this file. So i think we either
1. keep the functions as they are and have consistency on the file level, or
2. change both this function and the other `retry`s to be consistent with the above link, but break the API (or, maybe deprecate these and add a new set of functions)

What do you think?